wso2/identity-apps
# Use given SP's access URL for redirection in user invitation flow

### Purpose
> Give the capability to configure the service provider in email template used in "Invite user to set the password" flow and use that SP's access URL for redirection

### Related Issues
- https://github.com/wso2/product-is/issues/18939

### Steps to verify the fix
- Change the 'user-invitation(askPassword)' email template's user invitation link by adding spparam (ex: &sp=saml2-web-app-pickup-dispatch) similar to given below.
  ```
  href="{{account.recovery.endpoint-url}}/confirmrecovery.do?confirmation={{confirmation-code}}&amp;userstoredomain={{userstore-domain}}&amp;username={{url:user-name}}&amp;tenantdomain={{tenant-domain}}&amp;type=invite&sp=saml2-web-app-pickup-dispatch"
  ```
- Invite to a user to the organization.
- Open the user the invitation mail and setup a password to the account and finally can see below UI.

   ![access_URL](https://github.com/wso2/identity-apps/assets/36252606/818f22e4-a22f-4399-8698-f90c37da9dbd)

- Click the `Application` URl and it should redirect to  the given service provider's access URL.

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
